### ISSUE TITLE
ksh93 hang breaks one of automake's tests

### ISSUE THREAD

**Issue by meyering** [ksh93/ksh#316]
The following hangs both with Fedora 34's ksh-20120801-255 and with latest ksh93 from github:
```
(seq --format=z%g= 121; echo "eval ': \`(set) 2>&1\`'")|env -i ksh
```
yet it terminates if you change 121 to 120

It works as expected when using the latest ksh-2020 built from master.

**Comment by posguy99** [comment]
Doesn’t on Debian, either with current HEAD or with 93u+20120801-3.4+deb10u1.  Nor on Mint, with either HEAD or with 93u+20120801-7.

It *does* hang on AlmaLinux 8.3, which is more or less RH8, with current HEAD or with ksh-20120801-254.el8.x86_64.

--
Marc Wilson
***@***.***


On Sun, Jul 11, 2021, at 5:50 PM, meyering wrote:
> 
> 
> The following hangs both with Fedora 34's ksh-20120801-255 and with latest ksh93 from github:
> 
> `(seq --format=z%g= 121; echo "eval ': \`(set) 2>&1\`'")|env -i ksh
`

**Comment by McDutchie** [comment]
On my Mac (with GNU `seq` installed as `gseq`) I can get it to hang if the numeric argument is 1008 or higher:
```sh
(gseq --format=z%g= 1008; echo "eval ': \`(set) 2>&1\`'")|env -i arch/*/bin/ksh
```
Piping into ksh does not appear to be necessary to trigger the crash. The following easier to understand script also hangs on my system (on the last `eval` line):
```sh
for ((i=1; i<=1008; i++)); do
	eval "z$i="
done
eval ': `(set) 2>&1`'
```

**Comment by McDutchie** [comment]
Further experimentation shows that this is another bug with backtick command substitutions or perhaps the same bug as #305. The patch to fix #305 (which I'm still testing), which removes the separate mechanism for backtick comsubs, fixes this as well. It's about time to commit that patch I think.
